src/Streamer.cpp

```diff
@@ -48,102 +58,129 @@ FileWriter::Streamer::connect(std::string TopicName) {
     }
     // Error if the topic cannot be found in the metadata
     if (!Consumer->topicPresent(TopicName)) {
-      LOG(Sev::Error, "Topic {} not in broker, remove corresponding stream",
+      LOG(Sev::Error, "Topic \"{}\" not in broker, remove corresponding stream",
           TopicName);
-      return StreamerError::TOPIC_PARTITION_ERROR();
+      return {FileWriter::Status::StreamerStatus::TOPIC_PARTITION_ERROR,
+              nullptr};
     }
+    return {FileWriter::Status::StreamerStatus::WRITING, std::move(Consumer)};
   } catch (std::exception &Error) {
     LOG(Sev::Error, "{}", Error.what());
-    return StreamerError::CONFIGURATION_ERROR();
+    return {FileWriter::Status::StreamerStatus::CONFIGURATION_ERROR, nullptr};
   }
 
-  return StreamerError::WRITING();
+  return {FileWriter::Status::StreamerStatus::UNKNOWN_ERROR, nullptr};
```

Comment:
All exceptions are caught on line 67, so this line is unreachable anyway.

Reply:
I consider it good coding practice to always have a return statement on the last line of a function in  case something changes in the future. Furthermore, exceptions does not have to inherit from `std::exception`.